actionpack/lib/action_dispatch/http/url.rb

```diff
@@ -1,26 +1,36 @@
 # frozen_string_literal: true
 
 require "active_support/core_ext/module/attribute_accessors"
+require "action_dispatch/http/uri"
 
 module ActionDispatch
   module Http
     module URL
-      IP_HOST_REGEXP  = /\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$/
-      HOST_REGEXP     = /(^[^:]+:\/\/)?(\[[^\]]+\]|[^:]+)(?::(\d+$))?/
-      PROTOCOL_REGEXP = /^([^:]+)(:)?(\/\/)?$/
+      class << self
+        def tld_length=(tld_length)
+          ActionDispatch::Http::URI.tld_length = tld_length
+        end
 
-      mattr_accessor :secure_protocol, default: false
-      mattr_accessor :tld_length, default: 1
+        def tld_length
+          ActionDispatch::Http::URI.tld_length
+        end
+
+        def secure_protocol=(secure_protocol)
+          ActionDispatch::Http::URI.secure_protocol = secure_protocol
+        end
+
+        def secure_protocol
+          ActionDispatch::Http::URI.secure_protocol
+        end
 
-      class << self
         # Returns the domain part of a host given the domain level.
         #
         #    # Top-level domain example
         #    extract_domain('www.example.com', 1) # => "example.com"
         #    # Second-level domain example
         #    extract_domain('dev.www.example.co.uk', 2) # => "example.co.uk"
         def extract_domain(host, tld_length)
-          extract_domain_from(host, tld_length) if named_host?(host)
+          ActionDispatch::Http::URI.extract_domain(host, tld_length)
```

Comment:
As URL is a module and URI a class, I believe it would be confusing and not helpful to rename URI to URL. I am happy to deprecate the old static methods for URL.